lib/internal/url.js

```diff
@@ -105,7 +105,11 @@ function parse(input, base) {
       this[context].query = query;
       this[context].fragment = fragment;
       this[context].host = host;
-      this[searchParams] = new URLSearchParams(query);
+      if (this[searchParams]) {  // invoked from href setter
+        initSearchParams(this[searchParams], query);
```

Comment:
Actually, after looking at node_url.cc, it's not possible for there to be a question mark here, which means that these following lines are not needed for this case:

```js
init = String(init);
if (init[0] === '?') init = init.slice(1);
```

As a result of that, please move these lines from `initSearchParams` back to `URLSearchParams`'s constructor.